src/IO/ADIOS/ADIOS1IOHandler.cpp

```diff
@@ -977,7 +977,7 @@ ADIOS1IOHandlerImpl::writeAttribute(Writable* writable,
                                             "",
                                             getBP1DataType(parameters.dtype),
                                             nelems,
-                                            values.get());
+                                            values.get()); // Invalid read of size 1 (could be in ADIOS itself?)
```

Comment:
in `strlen()` in `adios_get_type_size (adios_bp_v1.c:2415)` in `adios_common_define_attribute_byvalue (adios_internals.c:1109)` in `adios_define_attribute_byvalue (adios.c:442)` in `openPMD::ADIOS1IOHandlerImpl::writeAttribute` on string value: `serial_fileBased_write%T`

Reply:
we already pass the invalid c-string.

`strlen((const char*)values.get())` is an invalid read of size 1

Reply:
Very likely because of https://github.com/openPMD/openPMD-api/pull/269#issuecomment-398033949.

Reply:
ah dang, just found it as well :)
yes, we can not defer/forward the pointer we get there :)